meshio/medit_io.py

```diff
@@ -118,13 +118,11 @@ def read_buffer(file):
 def write(filename, mesh):
     with open(filename, "wb") as fh:
         if mesh.points.dtype == c_float:
```

Comment:
This `if`-`else` could really be condensed into `{c_float: 1, c_double: 2}[mesh.points.dtype]`. No need for an assertion then either.

Reply:
I thought so too but ([above](https://github.com/nschloe/meshio/pull/292#discussion_r213922359)) hit a KeyError with `dtype('float64')`.  Tricky as 
```python
numpy.dtype('float64') == ctypes.c_double
```
but
```python
numpy.dtype('float64') not in {ctypes.c_double: None}
```
(I'm still scratching my head about that one.)

Reply:
That's really weird! Okay, best add a comment explaining that, otherwise I see myself bumping into the same issue in while when trying to "fix" it.

Reply:
Weirder:
```python
numpy.dtype(ctypes.c_double) in [ctypes.c_double]
```
but
```python
numpy.dtype(ctypes.c_double) not in {ctypes.c_double: 2}
```

Reply:
Your suggestion
```python
{c_float: 1, c_double: 2}[mesh.points.dtype]  # surprisingly fails
```
does work if the keys are wrapped with `dtype`:
```python
        version = {numpy.dtype(c_float): 1,
                   numpy.dtype(c_double): 2}[mesh.points.dtype]
```
That seems clear.  9dcf64bd5081be0dd37bf33731ecc25b8a95fe24